Mathlib/SetTheory/Ordinal/Notation/FundamentalSequence.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Data.Set.Operations
+
+/-!
+# Fundamental sequences
+
+If `x` is a successor limit in a linear order `α`, a fundamental sequence for `x` is a strictly
+monotonic sequence `ℕ → α` whose supremum is `x`.
+
+If `x` is not a successor limit, we can define two other types of fundamental sequences: the empty
+sequence `∅` with limit `⊥`, and the singleton sequence `{x}` with limit `succ x` (the limit here is
+more properly the least strict upper bound). This ensures that every countable ordinal
+(nonconstructively) has a fundamental sequence, and it simplifies the definition of the fast-growing
+hierarchy.
+
+To nicely manage these three different cases, we define `Sequence α = Option α ⊕ (ℕ → α)` as the
+type of sequences with length 0, 1, or `ω`. At the moment, the file contains only the basic API for
+this type.
```

Comment:
I wonder if it's more helpful to define this as an inductive type with three constructors, then you can give semantic names to the three options. Rather than `inl none`, `inl (some x)`, `inr f`, you'd have `Empty`, `Singleton`, `OfFun` or something...

Reply:
Would that give us the `{x} = Ø` and such lemmas for free? If so, then it definitely seems cleaner. If not, then really the fact that this is defined as a sum type is nothing but an implementation detail anyways.

Reply:
It wouldn't give those for free (I believe, since there's still the notation in the way), but it would mean that `ofFun` would be already given rather than a separate definition together with a simp lemma (ie `sum_inr_def` wouldn't need to exist). It'd also mean that recursion would have more meaningful names, such as in the def of `range`.

Reply:
Just did this change, and I do find things to be simpler. Thanks!